e2e-app/src/app/datepicker/focus/datepicker-focus.pw-spec.ts

```diff
@@ -0,0 +1,356 @@
+      await expectFocused(SELECTOR_DAY(new Date()), `Today's date should be focused`);
+    });
+  }
+
+  it(`should change month on click and keep 'next' arrow focused`, async() => {
+    await openDatepicker();
+
+    // first date of current month
+    const firstDate = getFirstOfMonth(new Date());
+    await test.page.waitForSelector(SELECTOR_DAY(firstDate));
+
+    // skipping one months
+    await test.page.click(SELECTOR_NEXT_MONTH);
+    await expectFocused(SELECTOR_NEXT_MONTH, `Next month arrow should be focused`);
+
+    // make sure we changed month
+    expect(await test.page.waitForSelector(SELECTOR_DAY(firstDate), {state: 'detached'}));
+  });
+
+  it(`should change month on click and keep 'prev' arrow focused`, async() => {
+    await openDatepicker();
+
+    // last date of current month
+    const lastDate = getLastOfMonth(new Date());
+    await test.page.waitForSelector(SELECTOR_DAY(lastDate));
+
+    // focus next month
+    await test.page.click(SELECTOR_PREV_MONTH);
+    await expectFocused(SELECTOR_PREV_MONTH, `Previous month arrow should be focused`);
+
+    await test.page.waitForSelector(SELECTOR_DAY(lastDate), {state: 'detached'});
+  });
+
+  it(`should re-focus current element after inside click`, async() => {
+    await openDatepicker();
+
+    // click on weekday header
+    await test.page.click(SELECTOR_FIRST_WEEKDAY);
+    await expectFocused(SELECTOR_DAY(new Date()), `Today's date should stay focused`);
+  });
+
+  if (process.env.BROWSER !== 'webkit') {
+    it(`should allow focusing datepicker input`, async() => {
+      await openDatepicker();
+
+      // focus input
+      await test.page.click(SELECTOR_DATEPICKER_INPUT);
+      await expectFocused(SELECTOR_DATEPICKER_INPUT, `Datepicker input should be focused`);
+
+      // tab should go back to datepicker
+      await sendKey(Key.Tab);
+      await expectFocused(SELECTOR_PREV_MONTH, `Previous Month arrow should be focused`);
+    });
+  }
+
+  it(`should be closed on Escape from input and keep focus`, async() => {
+    await openDatepicker();
+
+    // focus input
+    await test.page.click(SELECTOR_DATEPICKER_INPUT);
+    await expectFocused(SELECTOR_DATEPICKER_INPUT, `Datepicker input should be focused`);
+
+    // close
+    await sendKey(Key.ESC);
+    await test.page.waitForSelector(SELECTOR_DATEPICKER, {state: 'detached'});
+
+    // check input is still focused
+    await expectFocused(SELECTOR_DATEPICKER_INPUT, `Input element should stay focused after datepicker is closed`);
+  });
+
+  it(`should not allow any interactions when disabled`, async() => {
+    await disableDatepicker();
+    await preSelectDate();  // 10 AUG 2018
+    await openDatepicker();
+
+    await expectFocused(SELECTOR_TOGGLE, `Toggling element should stay focused`);
+
+    const dayElement = SELECTOR_DAY(new Date(2018, 7, 10));
+    await test.page.waitForSelector(dayElement);
+    let message = '';
+    try {
+      await test.page.click(dayElement, {timeout: 100});
```

Comment:
On window, webkit is much slower and is not enough to get the message `checking that element receives pointer events` (it fails with element still moving).

I tested it successfully on my laptop with 200, but I wonder it can introduce a flaky test, depending on the machine where it runs.

Reply:
I tried to use [waitForElementState](https://playwright.dev/docs/api/class-elementhandle?_highlight=waitforelementstate#elementhandlewaitforelementstatestate-options) before the click (to be sure that the element is not moving), but it doesn't help.

I guess it's something 'internally' which is slower, not giving the right message, so setting the timeout to 200 must be a good solution.

Reply:
ok